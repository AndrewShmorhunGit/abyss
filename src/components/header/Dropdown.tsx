import React from "react";
import styles from "@/app/styles/header.module.scss";
import { options } from "@/data/constants";
import { IScaleParams } from "@/interfaces/IScale";
import { useClickOutside } from "@/hooks/useClickOutside";

export function Dropdown({
  isScale,
  setScale,
  isActive,
  setIsActive,
}: IScaleParams) {
  const ref = React.useRef<HTMLDivElement | null>(null);
  useClickOutside(ref, () => setIsActive(false));

  if (isActive) {
    return (
      <div ref={ref} className={isActive ? styles.dropdown : styles.hide}>
        <div>
          {options.map((option: number, index: number) => {
            return (
              <button
                key={index}
                className={styles.btn + " " + styles.hover}
                style={{
                  fontWeight: option === +isScale[0] ? 800 : 500,
                }}
                onClick={() => {
                  setIsActive(false);
                  setScale([option + ""]);
                }}
              >
                {option === +isScale[0]
                  ? option + "%" + " " + "âœ”"
                  : option + "%"}
              </button>
            );
          })}
        </div>
      </div>
    );
  }
}
