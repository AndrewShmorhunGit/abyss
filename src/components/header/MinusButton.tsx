import { AiOutlineMinus } from "react-icons/ai";
import styles from "@/app/styles/header.module.scss";
import { decreaseScale } from "@/utils/functions";
import { IScaleOptions } from "@/interfaces/IScale";

export function MinusButton({ isScale, setScale }: IScaleOptions) {
  return (
    <button
      className={styles.btn}
      onClick={() => {
        setScale(decreaseScale(isScale));
      }}
    >
      <AiOutlineMinus />
    </button>
  );
}
