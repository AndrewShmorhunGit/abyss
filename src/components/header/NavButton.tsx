"use client";
import styles from "@/app/styles/header.module.scss";
import { useTranslateContext } from "@/providers/translate.context";
import { useState } from "react";
import { FaLocationArrow } from "react-icons/fa";

export function NavButton() {
  const { setPosition } = useTranslateContext();

  const [intervalId, setIntervalId] = useState<NodeJS.Timeout | null>(null);

  const startInterval = () => {
    if (!intervalId) {
      const newIntervalId = setInterval(() => {
        setPosition({ x: 0, y: 0 });
      }, 10);
      setIntervalId(newIntervalId);
    }
  };

  const stopInterval = () => {
    if (intervalId) {
      clearInterval(intervalId);
      setIntervalId(null);
    }
  };

  return (
    <button
      className={styles.btn}
      onClick={startInterval}
      onMouseLeave={stopInterval}
    >
      <FaLocationArrow />
    </button>
  );
}
