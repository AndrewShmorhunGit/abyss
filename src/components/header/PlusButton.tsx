import { AiOutlinePlus } from "react-icons/ai";
import styles from "@/app/styles/header.module.scss";
import { increaseScale } from "@/utils/functions";
import { IScaleOptions } from "@/interfaces/IScale";

export function PlusButton({ isScale, setScale }: IScaleOptions) {
  return (
    <button
      className={styles.btn}
      onClick={() => {
        setScale(increaseScale(isScale));
      }}
    >
      <AiOutlinePlus />
    </button>
  );
}
