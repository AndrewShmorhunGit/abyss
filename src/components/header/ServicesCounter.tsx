import styles from "@/app/styles/header.module.scss";
export function ServicesCounter() {
  return (
    <div className={`center ${styles.services}`}>
      <p>Services</p>
      <div className={styles.amount}>
        <span>0</span>
      </div>
    </div>
  );
}
