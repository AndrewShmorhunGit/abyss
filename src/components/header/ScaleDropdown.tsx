"use client";
import React from "react";
import styles from "@/app/styles/header.module.scss";
import { Dropdown } from "./Dropdown";
import { useScaleContext } from "@/providers/scale.context";

export function ScaleDropdown() {
  const { isActive, setIsActive, isScale, setScale } = useScaleContext();
  return (
    <div>
      <button
        className={styles.btn}
        onClick={() => (isActive ? "" : setIsActive(true))}
      >
        {isScale}%
      </button>
      {isActive && (
        <Dropdown
          isScale={isScale}
          setScale={setScale}
          isActive={isActive}
          setIsActive={setIsActive}
        />
      )}
    </div>
  );
}
