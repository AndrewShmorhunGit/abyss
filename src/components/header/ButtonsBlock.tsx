import { links } from "@/data/constants";
import { ListButton } from "./ListButton";
import { MinusButton } from "./MinusButton";
import { NavButton } from "./NavButton";
import { PlusButton } from "./PlusButton";
import { ScaleDropdown } from "./ScaleDropdown";
import { MainButtonsWrapper, ButtonsWrapper } from "./Wrappers";
import Link from "next/link";
import { ThemeButton } from "./ThemeButton";

function Links() {
  return (
    <div className="center gap" style={{ gap: "1.2rem" }}>
      {links.map(({ icon, link }) => {
        return (
          <Link
            key={link}
            href={link}
            target="_blank"
            rel="noopener noreferrer"
          >
            {icon}
          </Link>
        );
      })}
    </div>
  );
}

export function ButtonsBlock() {
  return (
    <MainButtonsWrapper>
      <ThemeButton />
      <Links />
      <ButtonsWrapper>
        <ListButton />
        <NavButton />
      </ButtonsWrapper>
      <ButtonsWrapper>
        <MinusButton />
        <ScaleDropdown />
        <PlusButton />
      </ButtonsWrapper>
    </MainButtonsWrapper>
  );
}
