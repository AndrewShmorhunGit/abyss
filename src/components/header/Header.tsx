import { ServicesCounter } from "./ServicesCounter";
import { HeaderWrapper } from "./containers/Wrappers";
import React from "react";
import { ButtonsBlock } from "./buttons/ButtonsBlock";

export function Header() {
  return (
    <HeaderWrapper>
      <ServicesCounter />
      <ButtonsBlock />
    </HeaderWrapper>
  );
}
