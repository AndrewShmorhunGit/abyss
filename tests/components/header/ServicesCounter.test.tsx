import { ServicesCounter } from "@/components/header/ServicesCounter";
import React from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";

// Mock the styles import
jest.mock("@/app/styles/header.module.scss", () => ({
  services: "mocked-services-class",
  amount: "mocked-amount-class",
}));

describe("ServicesCounter Component", () => {
  it("renders the component with the expected structure", () => {
    const { container } = render(<ServicesCounter />);

    // Check if the container has the "center" class
    expect(container.firstChild).toHaveClass("center");

    // Check if the container has the "mocked-services-class" class
    expect(container.firstChild).toHaveClass("mocked-services-class");

    // Check if the "Services" text is present
    expect(container).toHaveTextContent("Services");

    // Check if the amount container has the "mocked-amount-class" class
    const amountContainer = container.querySelector(".mocked-amount-class");
    expect(amountContainer).toBeInTheDocument();

    // Check if the "0" text is present inside the amount container
    expect(amountContainer).toHaveTextContent("0");
  });

  it("matches the snapshot", () => {
    const tree = renderer.create(<ServicesCounter />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
