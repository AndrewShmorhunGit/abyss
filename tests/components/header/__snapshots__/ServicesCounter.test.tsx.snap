// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServicesCounter Component matches the snapshot 1`] = `
<div
  className="center mocked-services-class"
>
  <p>
    Services
  </p>
  <div
    className="mocked-amount-class"
  >
    <span>
      0
    </span>
  </div>
</div>
`;

exports[`matches the snapshot 1`] = `
<div
  className="center services"
>
  <p>
    Services
  </p>
  <div
    className="amount"
  >
    <span>
      0
    </span>
  </div>
</div>
`;
